File: API/Entities/Ship.cs
namespace API.Entities
{
    public class Ship
    {
        public int ID { get; set; }
        public string Numero { get; set; } // Au cas où le numéro contient des catactères alphanumériques
        public string Nom { get; set; }
        public short AnneeConstruction { get; set; }
        public float Longueur { get; set; }
        public float Largeur { get; set; }
        public int TonnageBrut { get; set; }
        public int TonnageNet { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public int UserId { get; set; }
        public User user { get; set; }

    }
}

File: client/src/app/_models/ship.ts
import { User } from "./user"

export interface Ship {
    id: number
    numero: string
    nom: string
    anneeConstruction: number
    longueur: number
    largeur: number
    tonnageBrut: number
    tonnageNet: number
    createdAt: string
    userId: number
    user: User
  }